### ISSUE TITLE
Migrate internal `ops.*` calls to `backend.ops.*`

### ISSUE THREAD

**Issue by samudraneel05** [keras-team/keras#23696]
## Migrate internal `ops.*` calls to `backend.ops.*` where symbolic dispatch isn't needed
as mentioned in #23642, `backend.ops` is the preferred way to call ops in internal keras code. However, in the codebase, many internal paths still do `from keras.src import ops` and call public ops on concrete tensors.

`ops.*` wrappers exist for symbolic dispatch: they check `any_symbolic_tensors` and build graph nodes when given `KerasTensor`. On the concrete path that check is pure overhead, and the wrapper sometimes also performs validation/default handling that isn't needed internally. Calling `backend.ops.*` directly skips all of that.

Goal would be to replace `ops.*` with `backend.ops.*` in internal code paths that only ever see concrete tensors.

### Todo:

- [ ] `keras/src/layers/core`
- [ ] `keras/src/layers/attention`
- [ ] `keras/src/layers/convolutional`
- [ ] `keras/src/layers/merging`
- [ ] `keras/src/layers/normalization`
- [ ] `keras/src/layers/pooling`
- [ ] `keras/src/layers/preprocessing`
- [ ] `keras/src/layers/regularization`
- [ ] `keras/src/layers/reshaping`
- [ ] `keras/src/layers/rnn`
- [ ] `keras/src/metrics`
- [ ] `keras/src/optimizers` incl. `schedules`
- [ ] `keras/src/quantizers` incl. `modes`
- [ ] `keras/src/trainers` + `data_adapters`
- [ ] `keras/src/utils`
- [ ] `keras/src/visualization`
- [ ] Misc/ anything i missed

**Comment by samudraneel05** [comment]
EDIT: i'm starting work on this, however, this is a lot of work that can be broken into chunks and i might have missed out on some files or folders, so if anyone is interested in working on some other sections or files that have not been mentioned above please feel free to message up here with the section that you'd be interested in looking at.

**Comment by yeti107** [comment]
Hi @samudraneel05 ! I'd be very interested in working on this :) If you let me know what sections you/others would cover id be happy to cover some or all of the rest :) 

Let me know and I can start asap.

**Comment by Bratet** [comment]
hi @samudraneel05  I'd like to help too, i will take `keras/src/layers/attention` if it's still open, will open a PR once it's ready.

**Comment by samudraneel05** [comment]
hey @yeti107 yeti, you can look at some files/folders that i might've missed in my run-through, there are lots. @Bratet , same if that's all right for you.

**Comment by dhantule** [comment]
Hi @samudraneel05 
Thank you for opening this and for the community interest! Marking this for keras-team review.

**Comment by Bratet** [comment]
Hello @samudraneel05, following up on my earlier comment: I’ve opened draft PR #23716 for keras/src/layers/attention. Could you confirm this doesn’t overlap with your work? If the scope looks good, could someone assign me for this contribution ? Thanks

**Comment by yeti107** [comment]
Hi @samudraneel05 @dhantule 

I've completed the migration for the `keras/src/distillation` folder and opened a draft PR with the changes. All relevant tests are passing (16/16).

If everything looks good, could I please be assigned the issue so I can continue working through some remaining folders in line with any other contributors?

**Comment by samudraneel05** [comment]
hey @Bratet, I believe we are waiting for an analysis/review by maintainers as to whether this issue is a priority or not. with respect to your PR, instead of `Fixes #23720`, it will be better to just keep the body inside as 'Part of #23720' as there are a lot of other sub-sections and sub-issues under this issue. Currently, if your PR gets merged, it will close this entire issue. Apart from that, I have no immediate overlap with your PR. WRT further comments/code review, only maintainers will be able to help you with that. Cheers :)

**Comment by Bratet** [comment]
thanks for the update :) , @samudraneel05! just to clarify though : my attention PR is #23716, and its description already says 'Part of #23696'. #23720 is the separate distillation PR

**Comment by samudraneel05** [comment]
ahh my apologies, yepp, yours looks fine to me!
